### PR TITLE
Hts 1490 - implemented enter details page for secure journey

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala [hunk 1]
```diff
@@ -48,7 +49,7 @@ class StrideController @Inject() (val authConnector:       AuthConnector,
                                   messageApi:              MessagesApi)(implicit val transformer: NINOLogMessageTransformer)
```
Comment (sureshhmrc) [inline app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala]: my intellij complains about `implicit FrontendAppConfig` is not found - 
may be better move this `val frontendAppConfig:   FrontendAppConfig` to implicit declaration next to it ?

File: app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala [hunk 2]
```diff
@@ -137,24 +167,34 @@ class StrideController @Inject() (val authConnector:       AuthConnector,
           val piDisplayedToOperator = PersonalInformationDisplayed(nsiUserInfo.nino, nsiUserInfo.forename + " " + nsiUserInfo.surname, None, List.empty[String])
           auditor.sendEvent(PersonalInformationDisplayedToOperator(piDisplayedToOperator, operatorDetails, request.path), nsiUserInfo.nino)
 
-          Ok(views.html.customer_not_eligible(reason, nsiUserInfo))
+          Ok(views.html.customer_not_eligible(reason, Some(nsiUserInfo.forename → nsiUserInfo.surname), nsiUserInfo.nino, roleType))
+        }
+      },
+      whenIneligibleSecure = { (ineligible, nino, _) ⇒
+        IneligibilityReason.fromIneligible(ineligible).fold {
+          logger.warn(s"Could not parse ineligiblity reason: $ineligible")
```
Comment (sureshhmrc) [inline app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala]: typo ? `ineligiblity` -> `ineligibility`

File: conf/application.conf [hunk 3]
```diff
@@ -143,6 +143,7 @@ nino-logging.enabled = false
 
 stride {
   base64-encoded-roles        = [ "aHRzIGhlbHBkZXNrIGFkdmlzb3I=" ]
+  base64-encoded-secure-roles = [ "aHRzIGhlbHBkZXNrIGFkdmlzb3Igc2VjdXJl" ]
```
Comment (sureshhmrc) [inline conf/application.conf]: is the below config still required ? 

```
manual-account-creation {
   enabled = true
 }
```

File: test/uk/gov/hmrc/helptosavestridefrontend/controllers/StrideControllerSpec.scala [hunk 4]
```diff
@@ -231,13 +254,30 @@ class StrideControllerSpec
...
 
           val result = controller.customerNotEligible(fakeRequestWithCSRFToken)
           status(result) shouldBe OK
           contentAsString(result) should include("Customer is not eligible for a Help to Save account")
+          contentAsString(result) should include("Name:")
+          contentAsString(result) should include("Create account manually")
+        }
+      }
+
+      "show the you-are-not-eligible page if session is found in mongo and but user is NOT eligible and the role type is secure" in {
```
Comment (sureshhmrc) [inline test/uk/gov/hmrc/helptosavestridefrontend/controllers/StrideControllerSpec.scala]: `show the customer-not-eligible page........` ?

File: conf/application.conf (hunk 3 above)
Comment (sureshhmrc) [reply conf/application.conf]: so, can we delete it now, if its no longer required ?

Comment (AndyHWChung) [reply conf/application.conf]: I thought it was deleted..let me take a look

Comment (AndyHWChung) [reply conf/application.conf]: nope, you're right - it's still there. Just deleted and pushed.

**Pull request merged**